### DISCUSSION TITLE
Passing logging to QueryBuilder

### DISCUSSION THREAD

**Discussion by wenerme** [mikro-orm/mikro-orm#5358] [General]
**Is your feature request related to a problem? Please describe.**

I want to enable the debug based on context, but I can not find a general way to preset the logger.enable for QueryBuild

**Describe the solution you'd like**

```
  protected async readBuilder(): Promise<{ builder: QueryBuilder<E> }> {
   
    const builder = this.repo.qb();
    let cond = await this.em.applyFilters<E>(this.Entity.name, {}, {}, 'read');
    cond && builder.andWhere(cond);
    if(debug){
        // builder.setLogger?
    }
    return { builder };
  }
```

**Describe alternatives you've considered**

always enable debug, do the context check in Logger

**Additional context**

https://mikro-orm.io/docs/logging

**Comment by B4nan** [comment]
The logger context can be passed when you create the QB, it's the last parameter of `em.createQueryBuilder()`:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/knex/src/SqlEntityManager.ts#L26

**Comment by B4nan** [comment]
779fa15ac762a9bae843a8b2f009e1861527469f
